XNAControls/XNAScrollPanel.cs

```diff
@@ -0,0 +1,658 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Input;
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Rampastring.Tools;
+
+namespace Rampastring.XNAUI.XNAControls;
+
+/// <summary>
+/// A panel that allows for scrolling.
+/// </summary>
+public class XNAScrollPanel : XNAPanel
+{
+    private const double SCROLL_REPEAT_TIME = 0.03;
+    private const double FAST_SCROLL_TRIGGER_TIME = 0.4;
```

Comment:
These constants are identical to the ones in `XNAListBox`. To enforce similarity of scrolling within XNAUI controls, it might be best to create a new `XNAUIConstants.cs` file for them and move them there as `public` constants, replacing the uses both here and in `XNAListBox`.